server/src/test/java/org/opensearch/action/search/SearchRequestOperationsListenerTests.java

```diff
@@ -0,0 +1,395 @@
+            public void onDFSPreQueryPhaseEnd(SearchPhaseContext context, long tookTime) {
+                assertEquals(timeInNanos.get(), tookTime);
+                assertNotNull(context);
+                dfsPreQueryPhaseEnd.incrementAndGet();
+            }
+
+            @Override
+            public void onCanMatchPhaseStart(SearchPhaseContext context) {
+                assertNotNull(context);
+                canMatchPhaseStart.incrementAndGet();
+            }
+
+            @Override
+            public void onCanMatchPhaseFailure(SearchPhaseContext context) {
+                assertNotNull(context);
+                canMatchPhaseFailure.incrementAndGet();
+            }
+
+            @Override
+            public void onCanMatchPhaseEnd(SearchPhaseContext context, long tookTime) {
+                assertNotNull(context);
+                canMatchPhaseEnd.incrementAndGet();
+            }
+
+            @Override
+            public void onQueryPhaseStart(SearchPhaseContext context) {
+                assertNotNull(context);
+                queryPhaseStart.incrementAndGet();
+            }
+
+            @Override
+            public void onQueryPhaseFailure(SearchPhaseContext context) {
+                assertNotNull(context);
+                queryPhaseFailure.incrementAndGet();
+            }
+
+            @Override
+            public void onQueryPhaseEnd(SearchPhaseContext context, long tookTime) {
+                assertNotNull(context);
+                queryPhaseEnd.incrementAndGet();
+            }
+
+            @Override
+            public void onFetchPhaseStart(SearchPhaseContext context) {
+                assertNotNull(context);
+                fetchPhaseStart.incrementAndGet();
+            }
+
+            @Override
+            public void onFetchPhaseFailure(SearchPhaseContext context) {
+                assertNotNull(context);
+                fetchPhaseFailure.incrementAndGet();
+            }
+
+            @Override
+            public void onFetchPhaseEnd(SearchPhaseContext context, long tookTime) {
+                assertNotNull(context);
+                fetchPhaseEnd.incrementAndGet();
+            }
+
+            @Override
+            public void onExpandSearchPhaseStart(SearchPhaseContext context) {
+                assertNotNull(context);
+                expandPhaseStart.incrementAndGet();
+            }
+
+            @Override
+            public void onExpandSearchPhaseFailure(SearchPhaseContext context) {
+                assertNotNull(context);
+                expandPhaseFailure.incrementAndGet();
+            }
+
+            @Override
+            public void onExpandSearchPhaseEnd(SearchPhaseContext context, long tookTime) {
+                assertNotNull(context);
+                expandPhaseEnd.incrementAndGet();
+            }
+        };
+
+        final List<SearchRequestOperationsListener> requestOperationListeners = new ArrayList<>(Arrays.asList(testListener, testListener));
```

Comment:
I see we have defined two listeners. Can't we randomize this as well? And then u don't have to hardcode value 2 below.